packages/configuration-svc/src/infrastructure/mongodb_configset_repo.ts

```diff
@@ -0,0 +1,349 @@
+        }
+
+        this._logger.info(`Successfully read file contents - globalConfigSet count: ${this._globalConfigSets.length}.`);
+    }
+    
+
+    private async _loadFromBcConfigSets():Promise<void>{
+        this._bcConfigSets.clear();
+        try{
+            const mongoBcConfigSets = await this._collectionBcConfigSets.find({}).toArray();
+
+            const configMap = new Map<string, BoundedContextConfigurationSet[]>();
+            mongoBcConfigSets.forEach(doc => {
+                if (!configMap.has(doc.boundedContextName)) {
+                    configMap.set(doc.boundedContextName, []);
+                }
+                configMap.get(doc.boundedContextName)?.push(doc);
+            });
+
+        }catch (err) {
+            this._logger.error(err);
+            throw new Error("cannot load  MongoConfigSetRepo bcConfigsets");
+        }
+
+        this._logger.info(`Successfully fetch contents - bcConfigSets count: ${this._bcConfigSets.size}`);
+    }
+  
+
+
+    /**************************************
+     * BC config set code
+     ************************************/
+
+    private _deepCopyBoundedContextConfigSet(bcConfigSet: BoundedContextConfigurationSet): BoundedContextConfigurationSet {
+        // we need this to de-reference the objects in memory when passing them to callers
+        return JSON.parse(JSON.stringify(bcConfigSet));
+    }
+
+
+    private _configSetIdString(bcName: string): string {
+        return bcName.toUpperCase();
+    }
+
+    private _getBoundedContextConfigVersions(bcName: string): BoundedContextConfigurationSet[] {
+        const configs: BoundedContextConfigurationSet[] | undefined = this._bcConfigSets.get(this._configSetIdString(bcName));
+
+        if (!configs) {
+            return [];
+        }
+
+        // sort by decreasing schemaVersion order (latest version first)
+        configs.sort((a: BoundedContextConfigurationSet, b: BoundedContextConfigurationSet) => semver.compare(b.schemaVersion, a.schemaVersion));
+
+        return configs;
+    }
+
+    private _getBoundedContextConfigLatestIteration(completeVersionConfigs: BoundedContextConfigurationSet[], schemaVersion: string): BoundedContextConfigurationSet | null {
+        const versionConfigSets: BoundedContextConfigurationSet[] = completeVersionConfigs.filter(value => value.schemaVersion === schemaVersion);
+
+        if (versionConfigSets.length <= 0) {
+            return null;
+        }
+
+        versionConfigSets.sort((a: BoundedContextConfigurationSet, b: BoundedContextConfigurationSet) => b.iterationNumber - a.iterationNumber);
+        const lastIteraction = versionConfigSets[0];
+
+        return lastIteraction ?? null;
+    }
+
+    async fetchAllBoundedContextConfigSets(): Promise<BoundedContextConfigurationSet[]> {
+        const allVersions: BoundedContextConfigurationSet[] = [];
+
+        for (const key of this._bcConfigSets.keys()) {
+            const versions = this._bcConfigSets.get(key) || [];
```

Comment:
Could use ?? instead of ||. 
const versions = this._bcConfigSets.get(key) ?? [];